gpt4all-chat/src/chatmodel.h

```diff
@@ -16,20 +16,39 @@
 #include <Qt>
 #include <QtGlobal>
 
+struct PromptAttachment {
+    Q_GADGET
+    Q_PROPERTY(QUrl url MEMBER url)
+    Q_PROPERTY(QString file MEMBER file)
+
+public:
+    QUrl url;
+    QString file;
+    bool operator==(const PromptAttachment &other) const {
+        return url == other.url &&
+               file == other.file;
+    }
+    bool operator!=(const PromptAttachment &other) const {
+        return !(*this == other);
+    }
```

Comment:
This is not necessary since at least C++20: https://en.cppreference.com/w/cpp/language/default_comparisons